Alignment/OfflineValidation/python/overlapValidationPlot.py

```diff
@@ -0,0 +1,250 @@
+from __future__ import absolute_import
+from __future__ import print_function
+
+from builtins import range
+import math
+
+import ROOT
+
+from .TkAlStyle import TkAlStyle
+
+dirNameList=["z","r","phi"]# in general directions are labeled z=0 r =1 phi =2 throughout this, I should probably think of something more elegant
+detNameList = ("BPIX", "FPIX", "TIB", "TID", "TOB", "TEC")
+
+def hist(tree_file_name, hist_name,subdet_ids,module_directions,overlap_directions,profile_directions):
+    f = ROOT.TFile(tree_file_name)
+    t = f.Get("analysis/Overlaps")
+    
+    dimList=[[30,0,20],[60,0,20],[70,20,55],[110,20,55],[110,55,110],[280,20,110]] #Dimension of tracker in cm for each subdet [z,r1,r2]
+    h = []
+    for subdet in range(6):#Creates a 4-D list of empty histograms for permutations of subdetector, overlap direction, module direction and profile direction.
+		h.append([])
+		for module in range(3):
+			h[subdet].append([])
+			for overlap in range(3):
+				h[subdet][module].append([])
+				for profile in range(4):
+					if (subdet+1 == 1 and (module == 1 or overlap ==1))or (subdet+1== 2 and (module == 0 or overlap == 0))or ((subdet+1== 3 or subdet+1== 5) and (overlap != 2 or module == 1))or ((subdet+1== 4 or subdet+1== 6) and (overlap != 2 or module == 0)):
```

Comment:
Given this condition check repeats several times, it would be efficient and less buggy to have a function or flag to run the check.

Reply:
Alright, I've just changed this check to be a function.